app/views/admin/OutcomesReportView.coffee

```diff
@@ -192,66 +198,82 @@ OutcomesReportComponent = Vue.extend
+            @fetchClassrooms(teacher).then (classrooms) =>
+              @classrooms = classrooms
+              Promise.all @classrooms.map (classroom) =>
+                @fetchStudentSessions(classroom).then (sessions) =>
+                  Vue.set classroom, 'sessions', sessions
+            Promise.all([
+              @fetchCourseInstances(teacher).then (courseInstances) =>
+                @courseInstances = courseInstances
+              @fetchCourses().then (courses) =>
+                @courses = courses
+            ]).then ([courseInstances, courses]) =>
+              courseIDs = _.uniq courseInstances.map (courseInstance) =>
+                courseInstance.courseID
+              indexedCourses = _.indexBy(courses, '_id')
+              @courses = utils.sortCourses(courseIDs.map (courseID) =>
+                indexedCourses[courseID]
+              )
+          ])
     
     fetchCompleteUser: (data) ->
-      user = new User(data)
-      console.log data
-      user.fetch()
-      user.once 'sync', (fullData) =>
-        @teacher = fullData.toJSON()
-        console.log @teacher
-        @fetchTrialRequest()
-        @fetchClassrooms()
-        @fetchCourses()
+      new Promise (accept, reject) ->
+        user = new User(data)
+        user.fetch().then (data) ->
+          return accept(data)
+        , (error) ->
+          return reject(error) if error
     
-    fetchTrialRequest: ->
-      trialRequests = new TrialRequests()
-      trialRequests.fetchByApplicant(@teacher._id)
-      trialRequests.once 'sync', =>
-        if trialRequests.length is 0
-          noty text: "WARNING: No trial request found for that user!", type: 'error'
-        @trialRequest = trialRequests.models[0]?.toJSON()
+    fetchTrialRequest: (teacher) ->
+      new Promise (accept, reject) ->
+        trialRequests = new TrialRequests()
+        trialRequests.fetchByApplicant(teacher._id).then (data) ->
+          if data.length is 0
+            noty text: "WARNING: No trial request found for that user!", type: 'error'
+            return reject("WARNING: No trial request found for that user!")
+          return accept(data)
+        , (error) ->
+          return reject(error) if error
 
-    fetchClassrooms: ->
-      classrooms = new Classrooms()
-      classrooms.fetchByOwner(@teacher._id)
-      classrooms.once 'sync', =>
-        @classrooms = classrooms.toJSON()
-        @fetchStudentSessions()
-        @fetchCourseInstances()
+    fetchClassrooms: (teacher) ->
+      new Promise (accept, reject) ->
+        classrooms = new Classrooms()
+        classrooms.fetchByOwner(teacher._id).then (data) ->
+          accept(data)
+        , (error) ->
+          reject(error)
+
+    fetchCourseInstances: (teacher) ->
+      new Promise (accept, reject) ->
+        courseInstances = new CourseInstances()
+        courseInstances.fetchByOwner(teacher._id).then (data) ->
+          accept(data)
+        , (error) ->
+          reject(error)
```

Comment:
Functions like this should at least be unnecessary with the new api module. It'd probably just be `api.courseInstances.fetchByOwner(teacher._id)` and that returns the promise. In the near future...